router/router.go

```diff
@@ -315,6 +330,42 @@ func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *R
 	return r, nil
 }
 
+func applyBidderInfoConfigOverrides(bidderInfos config.BidderInfos, adaptersCfg map[string]config.Adapter) error {
+	for bidderName, bidderInfo := range bidderInfos {
+		if adapterCfg, exists := adaptersCfg[bidderName]; exists {
+			bidderInfo.Syncer = adapterCfg.Syncer.Override(bidderInfo.Syncer)
+
+			// comptability with legacy settings (if possible unambiguously)
+			if adapterCfg.UserSyncURL != "" {
+				if bidderInfo.Syncer == nil {
+					return fmt.Errorf("failed to apply legacy usersync_url setting for bidder %s, bidder does not define a user sync", bidderName)
+				}
+
+				endpointsCount := 0
+				if bidderInfo.Syncer.IFrame != nil {
+					bidderInfo.Syncer.IFrame.URL = adapterCfg.UserSyncURL
+					endpointsCount++
+				}
+				if bidderInfo.Syncer.Redirect != nil {
+					bidderInfo.Syncer.Redirect.URL = adapterCfg.UserSyncURL
+					endpointsCount++
+				}
+
+				if endpointsCount == 0 {
+					return fmt.Errorf("failed to apply legacy usersync_url setting for bidder %s, bidder does not define user sync endpoints", bidderName)
+				}
+
+				if endpointsCount > 1 {
```

Comment:
From the description: `Bidders may now specify both iframe and/or redirect endpoints.`

Given this, why is it an error for a bidder to specify both?

Reply:
This is not a general validation error. This logic only kicks in if an older style `PBS_ADAPTERS_FOO_USERSYNC_URL` override value is used. It is now possible for the bidder to have two endpoints, so which one is the override intended for? Rather than trying to guess, it's an error to provide an older-style user sync url override if the bidder defines more than one endpoint. The replacement format would be to include the type in the override key, such as `PBS_ADAPTERS_FOO_USERSYNC_IFRAME_URL`.

Reply:
Alternatively, I can omit this attempt at a compatibility layer and make it clear in the release notes this format is no longer supported.

Reply:
This makes sense now. You can only use the URL override if there is only one option. I was thinking the other way around, where we would only use the old style config if the new one is missing. Perhaps instead of checking for both new options, you could check which type the old URL was for, and only overwrite that setting in the new config.

One worry with this method is these old overrides sitting forgotten in the config in a legacy install, and someone having issues getting updated URL values to take. But given all the defaults have been removed, this will only occur when a custom usersync URL was set, and then the administrator forgets where they set the custom value.

All in all, I will say this can stay here. It is just migrating config location is always a pain.

Reply:
> Perhaps instead of checking for both new options, you could check which type the old URL was for, and only overwrite that setting in the new config.

The information is not represented in code. I would need to add a new yaml setting for something like "legacy sync type", which I'm not sure is worth it to just support this one case.

> One worry with this method is these old overrides sitting forgotten in the config in a legacy install

I share that concern. Perhaps it would be simpler to just throw an error if the old override format is used and ask hosts to update for this breaking config change.